### PR TITLE
feat: update esbuild to 0.16

### PR DESCRIPTION
**Pull request by silverwind**
See https://github.com/evanw/esbuild/releases/tag/v0.16.0

There is one breaking change with the legalComments default which changed to `none` and I just updated docs and test for this assuming you want to follow esbuild's default (I would).

### CODE REVIEW

**Comment by privatenumber** [comment]
I prefer to follow the default as well, but we'll need to avoid the breaking change to release this as a minor.

**Comment by silverwind** [comment]
Why not bump to 3.0.0? Want to include any other breaking changes in such a version?

**Comment by silverwind** [comment]
Ah, I see https://github.com/privatenumber/esbuild-loader/issues/275. Well to me personally, there is nothing critical in esbuild 0.16, so I'm happy to wait a bit longer until this is merged/released.

**Comment by servant-creations** [comment]
Any update on when this can be merged in? I would like to test esbuild-loader but am unable to. There is a [esbuild defect](https://github.com/evanw/esbuild/issues/2761) that was [recently fixed](https://github.com/evanw/esbuild/commit/46347f78f503a364eb5fc56d214a4fdb85be0324) and I can't test out esbuild-loader until it is in as a third party library I am using exposes this defect.

File: README.md [hunk 1]
```diff
@@ -338,7 +338,7 @@ Minify JS using equivalent but shorter syntax.
 #### legalComments
 Type: `'none' | 'inline' | 'eof'`
 
-Default: `'inline'`
```
Comment (privatenumber) [inline README.md:341]: This was reverted in https://github.com/evanw/esbuild/releases/tag/v0.16.10

**Comment by privatenumber** [comment]
@MonkBen 
Happy to move forward with this given your reason.

Can either of you upgrade esbuild to [v0.16.13](https://github.com/evanw/esbuild/releases/tag/v0.16.13) and address the `legalComments` default?

**Pull request closed without merging**